src/persist-client/src/write.rs

```diff
@@ -123,7 +123,10 @@ where
     #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
     pub async fn fetch_recent_upper(&mut self) -> Antichain<T> {
         trace!("WriteHandle::fetch_recent_upper");
-        self.machine.fetch_upper().await
+        // TODO: Do we even need to track self.upper on WriteHandle or could
+        // WriteHandle::upper just get the one out of machine?
+        self.upper = self.machine.fetch_upper().await;
+        self.upper.clone()
```

Comment:
Don't be randomized by this, but wanted to offer instead
```rust
let fresh_upper = self.machine.fetch_upper().await;
self.upper.clone_from(&fresh_upper);
fresh_upper
```

Reply:
done!

persist should probably do a more general pass on Antichain ownership and clone discipline at some point. it's definitely one of the areas where we're currently sloppy